src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTagger.cs

```diff
@@ -154,7 +155,8 @@ public IReadOnlyList<TagSpan<IntraTextAdornmentTag>> GetTags(NormalizedSnapshotS
                     // active view and disregards and requests for tags not in that view
                     var fullSpan = new SnapshotSpan(snapshot, 0, snapshot.Length);
                     var tags = _tagAggregator.GetTags(new NormalizedSnapshotSpanCollection(fullSpan));
-                    foreach (var tag in tags)
+                    var uniqueTags = tags.GroupBy(tag => tag.Tag.Hint.Span.Start).Select(group => group.First());
+                    foreach (var tag in uniqueTags)
                     {
                         // Gets the associated span from the snapshot span and creates the IntraTextAdornmentTag from the data
                         // tags. Only dealing with the dataTagSpans if the count is 1 because we do not see a multi-buffer case
```

Comment:
commenting about the code two lines below.  we have the tag's span.  why are we then mapping it back to the very snapshot we compute the tagspan for in the first place?